markdown/renderer_heading.go

```diff
@@ -0,0 +1,59 @@
+package markdown
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/mattn/go-runewidth"
+	"github.com/yuin/goldmark/ast"
+)
+
+func (r *render) renderHeading(node *ast.Heading) error {
+	underlineHeading := false
+	if r.mr.hasUnderlineHeadings() {
+		underlineHeading = node.Level <= 2
+	}
+
+	if !underlineHeading {
+		r.w.Write(bytes.Repeat([]byte{'#'}, node.Level))
+		r.w.Write(spaceChar)
+	}
+
+	var headBuf bytes.Buffer
+	headBuf.Reset()
+
+	for n := node.FirstChild(); n != nil; n = n.NextSibling() {
+		if err := ast.Walk(n, func(inner ast.Node, entering bool) (ast.WalkStatus, error) {
+			if entering {
+				if err := ast.Walk(inner, r.mr.newRender(&headBuf, r.source).renderNode); err != nil {
+					return ast.WalkStop, err
+				}
+			}
+			return ast.WalkSkipChildren, nil
+		}); err != nil {
+			return err
+		}
+	}
+
+	id, hasId := node.AttributeString("id")
+	if hasId {
+		_, _ = fmt.Fprintf(&headBuf, " {#%s}", id)
+	}
+
+	_, _ = r.w.Write(headBuf.Bytes())
+
+	if underlineHeading {
```

Comment:
You were able to compose two things together and reuse a lot, nice! 💪🏽

Reply:
I just copied all that from the way you do table, haha.

I wonder what happens if you put header in a table cell. If stuff won't break. But who does that...

Reply:
Oh, it's actually impossible to put headers inside a table cell. So it's OK. :D

Reply:
👍🏽